include/NeoFOAM/core/Dictionary.hpp

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include <unordered_map>
+#include <any>
+#include <string>
+
+namespace NeoFOAM
+{
+
+class Dictionary
+{
+public:
+
+    Dictionary() = default;
+    // Dictionary(Dictionary&);
+    // Dictionary(std::unordered_map<std::string, std::any> data);
+
+    void insert(const std::string& key, const std::any& value);
+    std::any& operator[](const std::string& key);
+    const std::any& operator[](const std::string& key) const;
+
+    template<typename T>
+    T& get(const std::string& key)
+    {
+        return std::any_cast<T&>(data_.at(key));
+    }
+
+    template<typename T>
+    const T& get(const std::string& key) const
+    {
+        return std::any_cast<const T&>(data_.at(key));
+    }
+
+    Dictionary& subDict(const std::string& key);
+
+private:
```

Comment:
If we provide direct access to the map we use the erase and other functions of map until we add convinience wrapper.

```suggestion

std::unordered_map<std::string, std::any> & get_map() const {return data_;};

const std::unordered_map<std::string, std::any> & get_map() const {return data_;};

private:
```